From 7db93b7cd7ab860eb05e5267797e24430caa918a Mon Sep 17 00:00:00 2001
From: Heiko Becker <heiko.becker@kde.org>
Date: Thu, 21 Jan 2021 22:00:33 +0100
Subject: [PATCH 1/2] Replace uint with unsigned int

Improves portability by avoiding the non-standard 'uint'. For example
this fixes building with musl libc:

".../kwin_wrapper.c:45:9: error: unknown type name 'uint';
    did you mean 'int'?"
---
 helpers/wayland_wrapper/kwin_wrapper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/helpers/wayland_wrapper/kwin_wrapper.c b/helpers/wayland_wrapper/kwin_wrapper.c
index 061e05c06..6cd221da7 100644
--- a/helpers/wayland_wrapper/kwin_wrapper.c
+++ b/helpers/wayland_wrapper/kwin_wrapper.c
@@ -42,7 +42,7 @@ pid_t launch_kwin(struct wl_socket *socket, int argc, char **argv)
         snprintf(fdString, sizeof(fdString) - 1, "%d", wl_socket_get_fd(socket));
 
         char **args = calloc(argc + 6, sizeof(char *));
-        uint pos = 0;
+        unsigned int pos = 0;
         args[pos++] = (char *)"kwin_wayland"; //process name is the first argument by convention
         args[pos++] = (char *)"--wayland_fd";
         args[pos++] = fdString;
-- 
GitLab


From 44713ac64280b5e48b47199dace333b128920b73 Mon Sep 17 00:00:00 2001
From: Heiko Becker <heiko.becker@kde.org>
Date: Thu, 21 Jan 2021 22:18:33 +0100
Subject: [PATCH 2/2] Include <fcntl.h> for open()

According to POSIX [1] that's where it is defined.

[1] https://pubs.opengroup.org/onlinepubs/009604499/basedefs/fcntl.h.html
---
 helpers/wayland_wrapper/wl-socket.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/helpers/wayland_wrapper/wl-socket.c b/helpers/wayland_wrapper/wl-socket.c
index 4110e9529..34eb59305 100644
--- a/helpers/wayland_wrapper/wl-socket.c
+++ b/helpers/wayland_wrapper/wl-socket.c
@@ -10,6 +10,7 @@
 
 #include <assert.h>
 #include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-- 
GitLab

